### ISSUE TITLE
Implement onQueryExecuted and onQueryFinished to update the PlayBtn state (sparnatural vs. Sparnatural ?)

### ISSUE THREAD

**Issue by tfrancart** [SteinerPascal/sparnatural-port#67]

**Comment by SteinerPascal** [comment]
Not sure if I fully comprehend the problem here. But as far as I see it, it shouldn’t be a problem. Let me propose some solutions:
- Prototype the HTMLElement (I would not recommend it. See: https://stackoverflow.com/questions/4670361/how-to-add-my-own-methods-to-htmlelement-object)
- Send a Callback or Boolean property with the onQueryUpdated like this:
```
onQueryUpdated: function(queryString, queryJson, onQuerySubmittedCallBackORProperty) {
            // ... here : query post-processing ...
            // then : integration with YASQE and YASR
            yasqe.setValue(queryString);
            yasqe.query();
           // when query finished and you would like the play btn to enable
          onSubmitCallBackOrProperty()
          // OR in case of a boolean var
         onSubmitCallBackOrProperty = true
}
```
- having a config property such as this (ENABLEPLAYBTN):
```
 // If you would like to overwride default settings
        let settings = {
          ENABLEPLAYBTN: true
          config: config,
          language: "fr",
          defaultEndpoint: function() { return $('#endpoint').text() },
          //language:lang,
          onQueryUpdated: function(queryString, queryJson, specProvider) {
              queryString = semanticPostProcess(queryString, queryJson);
              queryString = specProvider.expandSparql(queryString);
              yasqe.setValue(queryString);
            },
          // triggered when "play" button is clicked
          onSubmit: function(sparnat) {
            // trigger the query from YasQE
            yasqe.query();
            sparnat.onQuerySubmitted();
          }
        }
```

**Comment by SteinerPascal** [comment]
btw: I saw these lines as well
https://github.com/SteinerPascal/sparnatural-port/blob/04d26e0acdfe6ab025196c5447938cd5d4a88f84/src/sparnatural/components/Sparnatural.ts#L73-L79

Is that an attempt to fix this issue?

**Comment by tfrancart** [comment]
> Not sure if I fully comprehend the problem here.

**Comment by tfrancart** [comment]
> Not sure if I fully comprehend the problem here. 

The problem is that the HTML page must be able to call some functions of Sparnatural from the outside

Sparnatural emits the query but does not execute it and does not know when it is finished. The HTML page knows (through YasR), and when the query has finished executing, the HTML page needs to call onQueryFinished on Sparnatural. Similarly, Sparnatural may not always know when the query is executed, in case the "Play" button is not displayed in Sparnatural and it is the HTML page responsibility to execute the query.

So : the HTML page needs to be able to call onQuerySubmitted and onQueryFinished. Have a look at https://docs.sparnatural.eu/Javascript-integration#onsubmit-reference

How do I call these 2 functions - the ones you pointed at - from the HTML page ?

**Comment by SteinerPascal** [comment]
Yes okay then i understood the problem. Now the methods get called but the css animation has to be set accordingly. I also changed the naming a bit. 
And unfortunately 'un malheur' happened where the tree widget fixes are also included... I'm going to merge this now and then update the other branch...
Just have a look at the commit [60e78c6](https://github.com/SteinerPascal/sparnatural-port/commit/60e78c68b5ef73b7cb602c10b12ebae527fbfed3)

**Comment by SteinerPascal** [comment]
Btw. your question was also about sparnatural.ts vs Sparnatural.ts. I agree that the naming is unfortunate and i propose to change the first to SparnaturalElement and the latter to SparnaturalComponent. Since the first is a class extending from HTMLElement and the second is extending from the self defined HTMLComponent.

**Comment by tfrancart** [comment]
> i propose to change the first to SparnaturalElement and the latter to SparnaturalComponent. Since the first is a class extending from HTMLElement and the second is extending from the self defined HTMLComponent.

yes good idea
